File: wed 17 4 2024/9.rb
def self.set_ivehr_interoperability(flag)
    retVal = []
    text   = (flag.eql? true) ? 'Enable' : 'Disable'
    begin
       @logger.log_message("Login to Customer Account login", LogUtilities::INFO, false)
       create_lifeshield_instance
       @lifeshield.login
       retVal << eval("@lifeshield.#{@lifeshield.whereAmI}.clickAccountConfiguration")
       retVal << eval("@lifeshield.#{@lifeshield.whereAmI}.selectConnectivityManagement")
       nodeList  = eval("@lifeshield.#{@lifeshield.whereAmI}.getColumnValues('Name')")
       @logger.log_message("Connectivity Adapter Available : #{nodeList}  -- Input CA node name : #{NODE_NAME}", LogUtilities::INFO, false)
       nodeIndex = nodeList.find_index(NODE_NAME)
       @logger.log_message("Index of the node #{nodeIndex+1} && Node name #{NODE_NAME}", LogUtilities::INFO, false)
       retVal << eval("@lifeshield.#{@lifeshield.whereAmI}.selectRowOption(nodeIndex+1, 'View Details')")
       retVal << eval("@lifeshield.#{@lifeshield.whereAmI}.clickNodeSettingsEdit()")
       @logger.log_message("Edit button click in Node settings : #{retVal.last}", LogUtilities::INFO, false)
       condition = (flag.eql? false) ? '!' : ''
       retVal << eval("@lifeshield.#{@lifeshield.whereAmI}.enableInteroperability(#{flag})")
       options = eval("@lifeshield.#{@lifeshield.whereAmI}.getInteroperabilityConfigurationOptions") if (flag.eql? true)
       @logger.log_message("Available Configuration Option : #{options}", LogUtilities::INFO, false)
       retVal << eval("@lifeshield.#{@lifeshield.whereAmI}.selectInteroperabilityConfigurationOption(#{INTEROPERABILITY_NAME})") if (flag.eql? true)
       retVal << eval("@lifeshield.#{@lifeshield.whereAmI}.clickNodeSettingsSave()")
       @logger.log_message("Save button click in Node settings : #{retVal.last}", LogUtilities::INFO, false)
    rescue
       retVal << false
       @logger.log_message("Something went wrong in the method : #{__method__}", LogUtilities::ERROR, false)
       @logger.log_message("Exception caught: #{$!.inspect}\nError in method #{__method__}", LogUtilities::ERROR)
       @logger.log_message("Exception backtrace:\n #{$!.backtrace.join("\n")}", LogUtilities::DEBUG, false)
    ensure
       @lifeshield.close unless @lifeshield.nil?
       return retVal.present? && retVal.all?
    end
 end

File: wed 17 4 2024/9.py
def set_ivehr_interoperability(flag):
    retVal = []
    text = 'Enable' if flag else 'Disable'
    try:
        print("Login to Customer Account login", LogUtilities.INFO, False)
        create_lifeshield_instance()
        lifeshield.login()
        retVal.append(lifeshield.where_am_i.click_account_configuration())
        retVal.append(lifeshield.where_am_i.select_connectivity_management())
        nodeList = lifeshield.where_am_i.get_column_values('Name')
        print(f"Connectivity Adapter Available : {nodeList}  -- Input CA node name : {NODE_NAME}", LogUtilities.INFO, False)
        nodeIndex = nodeList.index(NODE_NAME)
        print(f"Index of the node {nodeIndex+1} && Node name {NODE_NAME}", LogUtilities.INFO, False)
        retVal.append(lifeshield.where_am_i.select_row_option(nodeIndex+1, 'View Details'))
        retVal.append(lifeshield.where_am_i.click_node_settings_edit())
        print(f"Edit button click in Node settings : {retVal[-1]}", LogUtilities.INFO, False)
        condition = '' if flag else '!'
        retVal.append(lifeshield.where_am_i.enable_interoperability(flag))
        options = lifeshield.where_am_i.get_interoperability_configuration_options() if flag else None
        print(f"Available Configuration Option : {options}", LogUtilities.INFO, False) if flag else None
        retVal.append(lifeshield.where_am_i.select_interoperability_configuration_option(INTEROPERABILITY_NAME)) if flag else None
        retVal.append(lifeshield.where_am_i.click_node_settings_save())
        print(f"Save button click in Node settings : {retVal[-1]}", LogUtilities.INFO, False)
    except Exception as e:
        retVal.append(False)
        print(f"Something went wrong in the method : set_ivehr_interoperability", LogUtilities.ERROR, False)
        print(f"Exception caught: {e}\nError in method set_ivehr_interoperability", LogUtilities.ERROR)
        print(f"Exception backtrace:\n{traceback.format_exc()}", LogUtilities.DEBUG, False)
    finally:
        if lifeshield:
            lifeshield.close()
        return bool(retVal) and all(retVal)
